src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -240,6 +239,70 @@ public ResponseEntity<Exam> updateExam(@PathVariable Long courseId, @RequestBody
         return ResponseEntity.ok(savedExam);
     }
 
+    /**
+     * PATCH /courses/{courseId}/exams/{examId}/working-time : Update the working time of all exams
+     *
+     * @param courseId          the course ID to which the exam belong to
+     * @param examId            the exam ID the working time should be extended for
+     * @param workingTimeChange the working time change in seconds (can be positive or negative, but must not be 0)
+     * @return the ResponseEntity with status 200 (OK) and with the updated exam as body
+     */
+    @PatchMapping("/courses/{courseId}/exams/{examId}/working-time")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Exam> updateExamWorkingTime(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody Integer workingTimeChange) {
+        log.debug("REST request to update the working time of exam with id {}", examId);
+
+        examAccessService.checkCourseAndExamAccessForInstructorElseThrow(courseId, examId);
+
+        if (workingTimeChange == 0) {
+            throw new BadRequestException();
+        }
+
+        var now = now();
+
+        // We have to get exercise groups as `scheduleModelingExercises` needs them
+        Exam exam = examService.findByIdWithExerciseGroupsAndExercisesElseThrow(examId);
+        var originalExamDuration = exam.getDuration();
+
+        // 1. Update the end date & working time of the exam
+        exam.setEndDate(exam.getEndDate().plusSeconds(workingTimeChange));
+        exam.setWorkingTime(exam.getWorkingTime() + workingTimeChange);
+        examRepository.save(exam);
+
+        // 2. Re-calculate the working times of all student exams
+        var studentExams = studentExamRepository.findByExamId(examId);
+        for (var studentExam : studentExams) {
+            Integer originalStudentWorkingTime = studentExam.getWorkingTime();
+            int originalTimeExtension = originalStudentWorkingTime - originalExamDuration;
+            // NOTE: take the original working time extensions into account
+            if (originalTimeExtension == 0) {
+                studentExam.setWorkingTime(originalStudentWorkingTime + workingTimeChange);
+            }
+            else {
+                double relativeTimeExtension = (double) originalTimeExtension / (double) originalExamDuration;
+                int adjustedTimeExtension = Math.toIntExact(Math.round(relativeTimeExtension * workingTimeChange));
+                studentExam.setWorkingTime(originalStudentWorkingTime + adjustedTimeExtension);
+            }
+            var savedStudentExam = studentExamRepository.save(studentExam);
+            // NOTE: if the exam already started, notify the student about the working time change
+            if (now.isAfter(exam.getVisibleDate())) {
```

Comment:
The comment refers to the start date, but you use the visibility date here.

Reply:
Updated the note. I guess there are already tasks running after visible date? Or is it enough to notify after start date?